google_reauth/reauth_creds.py

```diff
@@ -85,6 +85,46 @@ def from_OAuth2Credentials(cls, original):
         json = original.to_json()
         return cls.from_json(json)
 
+    def _refresh(self, http):
+        """Refreshes the access_token.
+
+        This method first checks by reading the Storage object if available.
+        If a refresh is still needed, it holds the Storage lock until the
+        refresh is completed.
+
+        Args:
+            http: an object to be used to make HTTP requests.
+
+        Raises:
+            HttpAccessTokenRefreshError: When the refresh fails.
+        """
+        if not self.store:
+            self._do_refresh_request(http)
+        else:
+            self.store.acquire_lock()
+            try:
+                new_cred = self.store.locked_get()
+
+                if (new_cred and not new_cred.invalid and
+                        new_cred.access_token != self.access_token and
+                        not new_cred.access_token_expired):
+                    _LOGGER.info('Updated access_token read from Storage')
+                    self._updateFromCredential(new_cred)
+                else:
+                    if (new_cred and not new_cred.invalid and
```

Comment:
these branches hurt my head. Can you factor out the common expressions into a higher-level if statement for these two?

Reply:
Sure.  I thought about that, but I also didn't want this to deviate *too* far from the _refresh code in oauth2client (this is a raw copy/paste, with the lines from 114 to 123 added in).  I could either refactor it or add a comment mentioning why the redundant checks are there?